deepchecks/tabular/context.py

```diff
@@ -175,8 +176,12 @@ def __init__(
             raise DeepchecksValueError('At least one dataset (or model) must be passed to the method!')
         if train is not None:
             train = Dataset.cast_to_dataset(train)
+            if train.name is None:
+                train.name = DEFAULT_DATASET_NAMES[0]
         if test is not None:
             test = Dataset.cast_to_dataset(test)
+            if test.name is None:
+                test.name = DEFAULT_DATASET_NAMES[1]
         # If both dataset, validate they fit each other
         if train and test:
```

Comment:
Should add validation making sure that both datasets don't have the same name (also in vision context)